lua/nui/popup/border.lua

```diff
@@ -143,17 +143,20 @@ local function calculate_buf_edge_line(internal, edge, text, align)
 
   local max_width = size.width - left_char:width() - right_char:width()
 
-  local content_text = Text(defaults(text, ""))
-  if mid_char:width() == 0 then
-    content_text:set(string.rep(" ", max_width))
+  local content_line
+  if type(text) == "function" then
+    content_line = text(max_width, mid_char)
```

Comment:
need to check the `content_line:width()` here and raise error if it's greater than `max_width`